rst_files/schelling.rst

```diff
@@ -266,10 +230,12 @@ then increase the number of agents.
         end
     end
 
-    println("Converged, terminating")
-    plot(plot_array..., layout=(count-1,1), size=(600,(count-1)*400))
+    plot(plot_array...,
+         layout = (length(plot_array), 1),
+         size = (600, 400 * length(plot_array)),
+         title = permutedims(string.("Cycle ", 1:length(plot_array))))
```

Comment:
If we are going to use permutedims, I think we need to move it to a separate line before the `plot` and have a comment explaining what it is doing.  Is it really worth it, though, or is there an easier thing to explain that may have an additional line of code or two?

Reply:
I think that we should hold off on using complicated code just for plots. We had a lot of `reshape`s and splatting in here as well.

A list comprehension probably works well for generating plot titles procedurally, e.g. `titles = ["cycle $n" for n in 1:num_plots]` or whatever

Reply:
`permutedims` is the with the `(2,1)` is same as `transpose` or `'`, but it works on nun-numerics (non-recursive). I think a one line comment should be able to communicate this.

Reply:
Yeah.  I think the general rule should be: if there is the word `permute` or `perm` in the julia function name, then we should ask whether it really is necessary.  Comprehensions or an extra line of code can make things easier to read.

Reply:
```
["cycle $n" for n in 1:length(plot_array)]
```
vs.
```
 permutedims(string.("Cycle ", 1:length(plot_array)))
```
... even with a comment, the former is easier to read.  In-built sorting functions, permutations, etc. are tricky and would require people to lookup how they work.

Reply:
The one above would not work. It would be something like
```
reduce(hcat, "cycle $n" for n in eachindex(plot_array))
```
which I would be fine with.

Reply:
`reduce` is discouraged and `reduce(hcat` should be avoided at all costs.  I would rather have 2-3 lines of code that are clear than trying to cram everything into a fancy function.

Reply:
Is `reshape` fine?
```
reshape(["cycle $n" for n in eachindex(plot_array)], 1, 5)
```

Reply:
I don't think it's useful; I would take it out and nothing changed.

Reply:
Eliminate the titles?

Reply:
I think your `reshape` is a good approach.  It is easy to know what the function does just by the title.

Reply:
@Nosferican Just check to make sure the `reshape` is necessary. I would take it out in some of the other lectures and it didn't break anything. 

Will go offline to work on another project for a bit, but I guess my big feedback here is that I think this is a lecture where the `mutable struct` isn't too bad of an idea.